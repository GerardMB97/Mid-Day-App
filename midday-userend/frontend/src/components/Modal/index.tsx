import colors from '../../../colors';
import React from 'react';
import { Text, StyleSheet, View } from 'react-native';

const styles = StyleSheet.create({
  modalContainer: {
    position: 'relative'
  },
  container: {
    width: 240,
    height: 50,
    backgroundColor: colors.alert,
    borderRadius: 10,
    padding: 5
  },
  triangle: {
    width: 30,
    height: 30,
    transform: [{ rotate: '45deg' }],
    backgroundColor: colors.alert,
    position: 'absolute',
    left: 20,
    top: -5
  }
});

export default function Modal ({ modalText }:{modalText:string}) {
  return (
    <View>
      <View style={styles.triangle}/>
      <Text style={styles.container}>{modalText}</Text>
    </View>
  );
}
