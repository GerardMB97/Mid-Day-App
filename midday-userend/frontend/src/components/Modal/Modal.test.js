import React from 'react';
import { render } from 'react-native-testing-library';
import Modal from '.';

describe('Given a Modal component', () => {
  describe('When rendered', () => {
    test('Then it should render correctly', () => {
      const tree = render(<Modal modalText='I"m a modal' ></Modal>);

      expect(tree).toMatchSnapshot();
    });
  });
});
