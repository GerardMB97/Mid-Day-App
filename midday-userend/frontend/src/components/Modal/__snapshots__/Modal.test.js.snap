// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Modal component When rendered Then it should render correctly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "rgb(247, 199, 48)",
        "height": 30,
        "left": 20,
        "position": "absolute",
        "top": -5,
        "transform": Array [
          Object {
            "rotate": "45deg",
          },
        ],
        "width": 30,
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "rgb(247, 199, 48)",
        "borderRadius": 10,
        "height": 50,
        "padding": 5,
        "width": 240,
      }
    }
  >
    I"m a modal
  </Text>
</View>
`;
